[NM,MC,AK] Add enhanced version of global-author-list (#586)
springernature/frontend-toolkits: toolkits/global/packages/global-author-list/README.md

**Example output**
`Author 1, Author 2 ... Author 5`

### Enhance with a toggle button which expands/truncates the list

#### Markup

```html
<div data-author-list>

>     <h3 data-author-list-heading>Authors</h3>

Reviewer: If the heading is here for accessibility reasons, should it not be included for the non-js example as well?

    <ul class="c-author-list">
        <li class="c-author-list__item">Author 1</li>
        <li class="c-author-list__item">Author 2</li>
        <li class="c-author-list__item">Author 3</li>
        <li class="c-author-list__item">Author 4</li>
        <li class="c-author-list__item">Author 5</li>
    </ul>
</div>
```

> ##### Heads up!
> 
> It is down to your application to decide whether or not to have the heading
> visibily hidden (it needs to be accessible to screen readers).

Reviewer: ```suggestion
> **NOTE**: It is down to your application to decide whether or not to have the heading
visibily hidden (it needs to be accessible to screen readers).
```

This is how we show "heads up" information

#### Javascript

> The `authorLists()` function comes in handy to initialise all lists present in the
> document. 

Reviewer: ```suggestion
The `authorLists()` function is used to initialise **all** lists present in the
document. 
```

```javascript
import {authorLists} from '@springernature/global-author-list/js';

authorLists(/* options */);
```

> For more flexibility you can use the `authorList()` function directly:

Reviewer: ```suggestion
For more flexibility you can target an individual author list by using the `authorList()` function directly:
```

```javascript
import {authorList} from '@springernature/global-author-list/js/author-list';

const myAuthorListContainer = document.querySelector('.my-author-list-container');

authorList(myAuthorListContainer/*, options*/).init();
```

#### Options

| Option            | Type   | Default value                | Description                                                              |
|-------------------|--------|------------------------------|--------------------------------------------------------------------------|
| headingSelector   | String | '[data-author-list-heading]' | Selector to the author list's heading element                            |
| listSelector      | String | 'ul'                         | Selector to the author list's list element                               |

> | authorHideClass   | String | 'c-author-list__hide'        | CSS class to hide items from the author list's list when it is truncated |

Reviewer: ```suggestion
| authorHideClass   | String | 'c-author-list__hide'        | CSS class to hide items from the list when it is truncated |
```

> | truncatedClass    | String | 'c-author-list--truncated'   | CSS class to toggle onto the list element when it is truncated           |

Reviewer: ```suggestion
| truncatedClass    | String | 'c-author-list--truncated'   | CSS class to indicate truncated state           |
```

> | listModifierClass | String | null                         | CSS class to add to the list the component is initialised                |

Reviewer: ```suggestion
| listModifierClass | String | null                         | CSS class to add on initialisation                |
```

| buttonClassList   | String | null                         | List of CSS classes to style the toggle button                           |

The data attribute options are the same, but are lowercase and hyphenated:

- `data-author-list-heading-selector`
- `data-author-list-list-selector`
- `data-author-list-author-hide-class`
- `data-author-list-truncated-class`
- `data-author-list-list-modifier-class`
- `data-author-list-button-class-list`

> Note: data attribute options will take precedence over any options set during initialisation.

Reviewer: ```suggestion
> **NOTE**: data attribute options will take precedence over any options set during initialisation.
```

##### Heads up!

- It is down to your application to toggle the `.js` class onto the `<html>`
element, if you use the default `authorHideClass` value.

> - The JS component make use of an SVG image as icon in the toggle button that
>   swaps from `+` to `-`. It is down to your application to include these `+` and
>   `-` icons in its SVG sprite.

Reviewer: ```suggestion
> **NOTE**: The JS component make use of an SVG image as icon in the toggle button that
  swaps from `+` to `-`. It is down to your application to include these `+` and
  `-` icons in its SVG sprite.
```

Reviewer 2: We are adding a link to the icons folder where the svg files are
